### PR TITLE
Implement effect of Cat Balou card #37

### PR DESCRIPTION
**Pull request by IvanovAndrew**

### CODE REVIEW

File: Bang/GameEvents/CardEffects/CatBalouActionHandler.cs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+            if (victim.Hand.Count + victim.ActiveCards.Count == 0)
+                return new ErrorState();
+
+            var codeToCard = new Dictionary<ClosedHandCard, BangGameCard>();
+            for (int i = 0; i < victim.Hand.Count; i++)
+            {
+                codeToCard[new ClosedHandCard {Code = i}] = victim.Hand[i];
+            }
+            
+            var response = new ChooseOneCardResponse
```
Comment (bashnia007) [inline Bang/GameEvents/CardEffects/CatBalouActionHandler.cs:22]: I think there is no difference between random dropping card from hand or giving a choice. But random is easier :)

File: Bang/GameEvents/Response.cs [hunk 2]
```diff
@@ -24,6 +26,23 @@ public class DefenceAgainstBang : Response
         public byte CardsRequired { get; set; }
     }
 
+    public class ForceCardToDrop : Response
```
[the remark this answers is not in the document]
Comment (IvanovAndrew) [reply Bang/GameEvents/Response.cs]: Renamed

File: Bang/GameEvents/CardEffects/CatBalouActionHandler.cs [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+using Bang.PlayingCards;
+
+namespace Bang.GameEvents.CardEffects
+{
+    internal class CatBalouHandler : CardActionHandler
+    {
+        public override HandlerState ApplyEffect(Game.Gameplay gameplay, Player victim, BangGameCard card)
+        {
+            if (victim.Hand.Count + victim.ActiveCards.Count == 0)
+                return new ErrorState();
```
[the remark this answers is not in the document]
Comment (IvanovAndrew) [reply Bang/GameEvents/CardEffects/CatBalouActionHandler.cs:13]: From now ErrorState has NotAllowedOperation side effect by default

File: Bang/GameEvents/CardEffects/CatBalouActionHandler.cs (hunk 1 above)
Comment (IvanovAndrew) [reply Bang/GameEvents/CardEffects/CatBalouActionHandler.cs:15]: Fixed

File: Bang.Tests/CatBalouSpecification.cs [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+using Xunit;
+
+using static Bang.Game.GamePlayInitializer; 
+
+namespace Bang.Tests
+{
+    public class CatBalouSpecification
+    {
+        [Fact]
+        public void After_played_Cat_Balou_card_goes_to_discard_pile()
```
[the remark this answers is not in the document]
Comment (IvanovAndrew) [reply Bang.Tests/CatBalouSpecification.cs:37]: Fixed

**Pull request merged**